mlir/lib/Conversion/TosaToMIOpen/TosaToMIOpen.cpp

```diff
@@ -229,10 +267,161 @@ class ConvConverter final : public OpConversionPattern<tosa::Conv2DOp> {
   }
 };
 
+class TransposeConverter final : public OpConversionPattern<tosa::TransposeOp> {
```

Comment:
I think the approach should be to backtrack the inputs and outputs of the tosa::Conv2DOp to see if it is written by a transpose. Then attach the transpose input and fix the layout. This would make the TransposeOp dead-code.

Reply:
We discussed offline and this will be sufficient for now. We will generalize transpose folding and fusion in a later PR. @jungpark-mlir please create a ticket and link to this PR.